src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -7177,29 +7177,27 @@ private StatementSyntax ParseEmbeddedStatement()
             // The consumers of embedded statements are expecting to receive a non-null statement 
             // yet there are several error conditions that can lead ParseStatementCore to return 
             // null.  When that occurs create an error empty Statement and return it to the caller.
-            StatementSyntax statement = this.ParseStatementCore() ?? SyntaxFactory.EmptyStatement(EatToken(SyntaxKind.SemicolonToken));
+            return CheckEmbeddedStatement(this.ParseStatementCore() ?? SyntaxFactory.EmptyStatement(EatToken(SyntaxKind.SemicolonToken)));
```

Comment:
>SyntaxFactory.EmptyStatement(EatToken(SyntaxKind.SemicolonToken)) [](start = 71, length = 65)

Please consider extracting the `ParseEmptyStatement` as in your earlier change.

Reply:
i'll do in followup :)   thanks!

Reply:
Done in this PR since i added the .aggressive bit as well.